### PR TITLE
Removes root dependency management

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Before, we imported BOMs at the root-most scope. This implies any
consumer of any library imports the same BOM. For example, I noticed
after clean installing brave-opentracing, that it was importing Netty's
BOM.

Root-level dependency management tempts us to accidentally add
dependencies to Zipkin's most core library. The convenience is not worth
the risk. Hence, this pushes out dependency management nearer to where
they are used.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
cc @reta

File: pom.xml [hunk 1]
```diff
@@ -155,233 +160,72 @@
...
-        <artifactId>zipkin-proto3</artifactId>
-        <version>0.2.2</version>
-      </dependency>
-      <dependency>
-        <groupId>com.squareup.wire</groupId>
-        <artifactId>wire-runtime</artifactId>
-        <version>${wire.version}</version>
-      </dependency>
-    </dependencies>
+    <!-- Be careful here, especially to not import BOMs as io.zipkin.zipkin2:zipkin has this parent.
```
Comment (codefromthecrypt) [inline pom.xml:163]: here's the important text

**Review by anuraaga — APPROVED** [review]

File: zipkin-collector/scribe/pom.xml [hunk 2]
```diff
@@ -50,6 +50,7 @@
     <dependency>
       <groupId>javax.annotation</groupId>
       <artifactId>javax.annotation-api</artifactId>
+      <version>1.3.1</version>
```
Comment (reta) [inline zipkin-collector/scribe/pom.xml:53]: Sorry, a bit late with my comment, wouldn't it be better to define `javax.annotation-api` version in the parent `pom.xml`? I see it being used in a few places.

Comment (codefromthecrypt) [reply zipkin-collector/scribe/pom.xml:53]: sure. want to raise a PR to add that property? and use it in the few?

**Pull request merged**